state/account.go

```diff
@@ -178,6 +187,16 @@ func (st *Account) PendingNonce() uint64 {
 	}
 }
 
+// PendingNonceConsideringFreshAccount return the pending nonce considering fresh legacy account
+func (st *Account) PendingNonceConsideringFreshAccount() uint64 {
```

Comment:
How about naming it `PendingNonce()` and renaming the original `LegacyPendingNonce()`

Reply:
`SetPendingNonce()` should also be updated for this change

Reply:
`PendingNonce` has too many uses in the code
if we want to do this, can do in a separate PR